### ISSUE TITLE
Error due to primaryKey in serializer

### ISSUE THREAD

**Issue by mattmcginnis** [danielspaniel/ember-data-factory-guy#198]
Adding a primaryKey in the profile serializer (for example) results in the following error:
```
You must include an 'id' for profile in an object passed to 'push'
```
See this [example commit](f18d71c72f2e96b537e39946d8759548fe30cf1a) that breaks tests.

Full repo [here](https://github.com/mattmcginnis/ember-data-factory-guy)

I can try to dig deeper and make the tests pass. Any hints appreciated.

**Comment by danielspaniel** [comment]
You've got me baffled as to what you are trying to accomplish exactly? .. So you set the primaryKey as 'primaryKey'  .. but that field does not exist in the model or the factory .. What is it you are trying to do?  why should it not break?

**Comment by mattmcginnis** [comment]
Sorry if it's not clear, I'll try to explain better.

From: http://emberjs.com/api/data/classes/DS.RESTSerializer.html#property_primaryKey

> The primaryKey is used when serializing and deserializing data. Ember Data always uses the id property to store the id of the record. The external source may not always follow this convention. In these cases it is useful to override the primaryKey property to match the primaryKey of your external store.

```javascript
//app/serializers/application.js
import DS from 'ember-data';

export default DS.JSONSerializer.extend({
  primaryKey: '_id'
});
```
When using a different primaryKey, you don't add "_id" as an attribute on the model, your server just returns json containing "_id" instead of "id".  It wasn't clear to me how to mock this in the tests, but adding the primaryKey to the serializer did create the error I see in my repo.

> Would you expect factory guy to make the id .. but use that key as the primary key ?? so it would be profileId instead of id? 

Yes, the idea is to make factory guy support a primaryKey other than "id".    Let me know if that makes sense.

**Comment by danielspaniel** [comment]
That was pretty silly of me not to see that .. ok .. I get you now.  
Can't believe no one ever brought this up. Maybe it's not used too often. 

 But never-the-less .. I am wondering how to hook this up. I will see what I can do.
  I don't suppose your interested in trying to fix it?  Might be fun exercise.

**Comment by mattmcginnis** [comment]
I'll give it a shot. If you have any intuition about where to look, let me know.

**Comment by danielspaniel** [comment]
I reckon the first issue is making the json. 
 The factories will make an id by default. 
 So when you build ... in the fixture converters ( there are 2 .. jsonapi and rest ) .. when your converting the raw fixture to the json used for the payload .. you would replace the id with that profileId ( or whatever the key name is ) 
 not really that hard. 

 That ( I think ) would do it.

**Comment by mattmcginnis** [comment]
The code to make this work is pretty trivial but there is some side-effect with the JSONAPISerializer that I don't understand. Maybe you can take a look.  There is one failing test with the following warning in the console: (related to the JSONAPI serializer)
```
Encountered "data" in payload, but no model was found for model name 
"datum" (resolved model name using dummy@serializer:cat:.modelNameFromPayloadKey("data"))
```

See this [commit](https://github.com/mattmcginnis/ember-data-factory-guy/commit/29503c7d538f7c2ef58a923c5a029dba384c4b32)

 [Full Repo](https://github.com/mattmcginnis/ember-data-factory-guy)

Basically what I did was identify the test that fails with a primaryKey specified and then make two new models (dog and cat which use the REST serializer and JSONAPI serializer respectively) and then copied the test that fails for each model.  Maybe it's overkill, I'm not sure.

**Comment by danielspaniel** [comment]
Can you make a pull request out of these commits? It's a bit easier for me to noodle around with it that way, and to bring your code into the main project ( if you know what I mean )

**Comment by mattmcginnis** [comment]
https://github.com/danielspaniel/ember-data-factory-guy/pull/199

**Comment by danielspaniel** [comment]
Everything looked ok by the way .. and more tests is always better than none.  
 One change I might make is to see if I could refactor it to have the fixtureConverter ( main ) class have a method that handled the overriden primaryKey value ..

**Comment by danielspaniel** [comment]
fixed with v2.5.6

**Comment by mattmcginnis** [comment]
Thanks for finishing this Daniel, much appreciated!
